src/modules/flow/http-client/http-client.c

```diff
@@ -0,0 +1,600 @@
+            "Empty response from %s", mdata->url);
+        return -EINVAL;
+    }
+
+    if (response->response_code != SOL_HTTP_STATUS_OK) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "%s returned an unknown response code: %d",
+            mdata->url, response->response_code);
+        return -EINVAL;
+    }
+
+    return 0;
+}
+
+static int
+check_key(const char *url, struct sol_json_token *token)
+{
+    char *ptr;
+
+    if (strncmp(url, "http://", sizeof("http://") - 1) == 0)
+        ptr = strstr(url + sizeof("http://") - 1, "/");
+    else
+        ptr = strstr(url, "/");
+    SOL_NULL_CHECK(ptr, -EINVAL);
+
+    return !sol_json_token_str_eq(token, ptr, strlen(ptr));
+}
+
+static void
+boolean_request_finished(void *data,
+    const struct sol_http_client_connection *connection,
+    struct sol_http_response *response)
+{
+    struct sol_flow_node *node = data;
+    struct http_data *mdata = sol_flow_node_get_private_data(node);
+    bool result = true;
+
+    if (check_response(mdata, node, connection, response) < 0)
+        return;
+
+    if (streq(response->content_type, "application/json")) {
+        struct sol_json_scanner scanner;
+        struct sol_json_token token, key, value;
+        enum sol_json_loop_reason reason;
+
+        sol_json_scanner_init(&scanner, response->content.data, response->content.used);
+        SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+            if (check_key(mdata->url, &key))
+                continue;
+            if (sol_json_token_get_type(&value) == SOL_JSON_TYPE_TRUE)
+                result = true;
+            else if (sol_json_token_get_type(&value) == SOL_JSON_TYPE_FALSE)
+                result = false;
+            else
+                goto err;
+            sol_flow_send_boolean_packet(node,
+                SOL_FLOW_NODE_TYPE_HTTP_CLIENT_BOOLEAN__OUT__OUT, result);
+            return;
+        }
+    } else {
+        if (!strncasecmp("true", response->content.data, response->content.used))
+            result = true;
+        else if (!strncasecmp("false", response->content.data, response->content.used))
+            result = false;
+        else
+            goto err;
+        sol_flow_send_boolean_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_CLIENT_BOOLEAN__OUT__OUT, result);
+        return;
+    }
+
+err:
+    sol_flow_send_error_packet(node, EINVAL,
+        "%s Could not parser the url's contents ", mdata->url);
+}
+
+static int
+boolean_post_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,
+    const struct sol_flow_packet *packet)
+{
+    int r;
+    bool b;
+    struct sol_http_param params;
+    struct http_data *mdata = data;
+    struct sol_http_client_connection *connection;
+    const struct http_client_node_type *type;
+
+    SOL_NULL_CHECK(mdata->url, -EINVAL);
+
+    r = sol_flow_packet_get_boolean(packet, &b);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sol_http_param_init(&params);
```

Comment:
the rest of this function should be a common one that receives the `value_as_string`. Then the only type-specific bit is the one that takes the packet and converts the value to string:

```c

   r = sol_flow_packet_get_boolean(packet, &b);
   SOL_INT_CHECK(r, < 0, r);
   return common_request(node, b ? "true" : "false");
}

```